cmd/edit/service/cmd.go

```diff
@@ -64,37 +54,72 @@ func init() {
 }
 
 func run(cmd *cobra.Command, argv []string) {
-	r := rosa.NewRuntime().WithOCM()
+	r := rosa.NewRuntime().WithOCM().WithFlagChecker()
 	defer r.Cleanup()
 
+	// Adding known flags to flag checker before parsing the unknown flags
+	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+		r.FlagChecker.AddValidFlag(flag)
+	})
+
+	err := arguments.ParseUnknownFlags(cmd, argv)
+	if err != nil {
+		r.Reporter.Errorf("Failed to parse flags: %v", err)
+		os.Exit(1)
+	}
+
+	if len(cmd.Flags().Args()) > 0 {
+		r.Reporter.Errorf("Unrecognized command line parameter")
+		os.Exit(1)
+	}
+
 	if args.ID == "" {
 		r.Reporter.Errorf("Service id not specified.")
 		cmd.Help()
 		os.Exit(1)
 	}
 
-	// Try to find the cluster:
+	// Try to find the service:
 	r.Reporter.Debugf("Loading service %q", args.ID)
 	service, err := r.OCMClient.GetManagedService(ocm.DescribeManagedServiceArgs{ID: args.ID})
 	if err != nil {
 		r.Reporter.Errorf("Failed to get service %q: %v", args.ID, err)
 		os.Exit(1)
 	}
 
-	parameters := service.Parameters()
-
-	if len(parameters) == 0 {
-		r.Reporter.Errorf("Service %q has no parameters to edit", args.ID)
+	// Setting parameter flags as valid
+	addOn, err := r.OCMClient.GetAddOn(service.Service())
+	if err != nil {
+		r.Reporter.Errorf("Failed to get add-on %q: %s", service.Service(), err)
 		os.Exit(1)
 	}
 
+	addonParameters := addOn.Parameters()
+	addonParameters.Each(func(param *cmv1.AddOnParameter) bool {
+		r.FlagChecker.AddValidParameter(param.ID())
+		return true
+	})
+
+	// Now that rosa knows the expected fields to validate,
+	// Validate that all of the user-specified flags are valid.
+	cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+		if !r.FlagChecker.IsValidFlag(flag) {
+			r.Reporter.Errorf("%q is not a valid flag", flag.Name)
+			os.Exit(1)
+		}
+	})
```

Comment:
This is also generic and perhaps shouldn't be here - if the FlagChecker has access to `cmd` (whether directly or via runtime) this could become something like `FlagChecker.ValidateFlags()`